src/attr/validators.py

```diff
@@ -63,6 +66,63 @@ def instance_of(type):
     return _InstanceOfValidator(type)
 
 
+@attrs(repr=False, frozen=True)
+class _MatchesReValidator(object):
+    regex = attrib()
+    flags = attrib()
+    func = attrib()
+
+    def __call__(self, inst, attr, value):
+        """
+        We use a callable class to be able to change the ``__repr__``.
+        """
+        if self.func is re.match:
```

Comment:
We should keep the call overhead as low as possible, so I would suggest to pass bound methods into this calls instead of a regex and a func and the dispatching.

Reply:
oh good idea -- this logic could be done in the constructor instead of at every call